src/tools/idlc/src/generator.c

```diff
@@ -395,12 +396,49 @@ idlc_generate(const idl_pstate_t *pstate, const idlc_generator_config_t *config)
   memset(&generator, 0, sizeof(generator));
   generator.path = file;
 
+  if(config->base_dir) {
+    char* rel_path = NULL;
+    char* old_dir = dir;
+    // If we run into a path error we can still recover and use existing dir
+    if(idl_relative_path(config->base_dir, pstate->sources->path->name, &rel_path) == IDLC_NO_MEMORY)
+      goto err_mkpath;
+    // If root is not a parent of file path, or there was an error resolving rel path
+    // TODO maybe find a common directory between the two and build from there?
+    if(rel_path == NULL || strncmp(rel_path, "..", 2) == 0)
```

Comment:
Should it not be: `rel_path` starts with `..` followed by a directory separator? It is quite the edge case that someone would seriously create a directory named `...` and I dare say it is unlikely anyone would run into it by accident!